sirepo/pkcli/job_agent.py

```diff
@@ -55,14 +57,48 @@ def default_command():
     pkdlog('{}', cfg)
     i = tornado.ioloop.IOLoop.current()
     d = _Dispatcher()
-    def s(n, x):
-        return i.add_callback_from_signal(d.terminate)
+    def s(*args):
+        return i.add_callback_from_signal(_terminate, d)
     signal.signal(signal.SIGTERM, s)
     signal.signal(signal.SIGINT, s)
     i.spawn_callback(d.loop)
     i.start()
 
 
+def start_sbatch():
+    def kill_agent(pid_file):
+        if socket.getfqdn() == pid_file.fqdn:
```

Comment:
pid_file.host would be clearer. I understand that getfqdn is being used here, but that's not the abstraction we really want to talk about. We use "host" everywhere else to mean "the name of a node".

socket.getfqdn() on my Mac nicely returns: `1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa`, which is useless. See https://stackoverflow.com/a/43330159 for discussion. It may return localhost, which is annoying. socket.gethostname() returns a fully qualified domain name on my Mac and on Linux/NERSC, it returns a name that is usable within the network context (possibly). 

Let's wrap this in a function that checks the result and gets the other. If gethostname does not return anything with a `.`, call getfqdn.

Reply:
Thanks for looking at this. Fix implemented.